tools/fregrid_acc/conserve_interp_acc.c

```diff
@@ -22,882 +22,606 @@
-        for(n=0; n<interp[m].nxgrid; n++) {
-          i2   = interp[m].i_out[n];
-          j2   = interp[m].j_out[n];
-          i1   = interp[m].i_in [n];
-          j1   = interp[m].j_in [n];
-          tile = interp[m].t_in [n];
-          area = interp[m].area [n];
-          nx1  = grid_in[tile].nx;
-          ny1  = grid_in[tile].ny;
-          n0 = j2*nx2+i2;
-          n1 = j1*nx1+i1;
-          if(cell_measures )
-            out_area[n0] += (area*field_in[tile].area[n1]/grid_in[tile].cell_area[n1]);
-          else
-            out_area[n0] += area;
-        }
-        for(i=0; i<nx2*ny2*nz; i++) {
-          if(field_out[m].data[i] != missing) {
-            i2 = i%(nx2*ny2);
-            field_out[m].data[i] *=  (out_area[i2]/grid_out[m].cell_area[i2]);
+#pragma acc parallel loop present(out_area[:ncells_output_grid])
+        for(int i=0; i<ncells_output_grid; i++) out_area[i] = 0.0;
+
+        for(int itile=0 ; itile<ntiles_input_grid; itile++) {
+          int ncells_input_grid = input_grid[itile].nxc * input_grid[itile].nyc;
+          Interp_per_input_tile *minterp_acc = NULL ; minterp_acc = interp_acc[otile].input_tile+itile;
+          double *p_gridin_area  = NULL ; p_gridin_area  = input_grid[itile].cell_area;
+          double *p_fieldin_area = NULL ; p_fieldin_area = field_in[itile].area;
+          int ixcells = minterp_acc->nxcells;
+#pragma acc parallel loop present(minterp_acc->output_parent_cell_index[:ixcells], \
+                                  minterp_acc->input_parent_cell_index[:ixcells], \
+                                  minterp_acc->xcell_area[:ixcells],         \
+                                  out_area[:ncells_output_grid])\
+                           copyin(p_fieldin_area[:ncells_input_grid],\
+                                  p_gridin_area[:ncells_input_grid])
+          for(int ix=0; ix<ixcells; ix++) {
+            int ij2 = minterp_acc->output_parent_cell_index[ix];
+            int ij1 = minterp_acc->input_parent_cell_index[ix];
+            double area = minterp_acc->xcell_area[ix];
+            if(cell_measures ) out_area[ij2] += (area*p_fieldin_area[ij1]/p_gridin_area[ij1]);
+            else out_area[ij2] += area;
+          }
+#pragma acc parallel loop present(p_fieldout_data[:ncells_output_grid], out_area[:ncells_output_grid], \
+                                  output_grid[otile].cell_area[:ncells_output_grid])
+          for(int i=0; i<ncells_output_grid; i++) {
+            if(p_fieldout_data[i] != missing)
+              p_fieldout_data[i] *=  (out_area[i]/output_grid[otile].cell_area[i]);
           }
         }
       }
     }
 
-    free(out_area);
-    free(out_miss);
-  }
+#pragma acc exit data copyout(p_fieldout_data[:ncells_output_grid])
+#pragma acc exit data delete(out_area[:ncells_output_grid],\
+                             out_miss[:ncells_output_grid])
+
+    free(out_area); out_area = NULL;
+    free(out_miss); out_miss = NULL;
+  } // otile
 
+  return;
```

Comment:
Is this supposed to be here?

Reply:
nope, I must added it while testing and forgot to remove it  :)